sponsorship_switzerland/models/completion_rules.py

```diff
@@ -131,6 +136,32 @@ def get_from_partner_ref(self, stmts_vals, st_line):
                 )
         return res
 
+    def recurring_contract_wire_transfer_mapping(self, stmts_val, st_line):
+        if "name" not in st_line:
+            return
+
+        name = st_line['name']
+        res = dict()
+
+        wire_transfer_mode = self.env['account.payment.mode'].search([
+            ("name", "=", "wire transfer")
+        ])
+
+        if not wire_transfer_mode:
+            logger.warning("Unable to find wire transfer payment mode")
+            return res
+
+        recurring_contract_groups = self.env['recurring.contract.group'].search([
+            "&",
+            ("bvr_reference", "!=", False),
+            ("payment_mode_id", "=", wire_transfer_mode.id)
+        ]).filtered(lambda it: name.startswith(it.bvr_reference))
+
+        if recurring_contract_groups:
+            res['partner_id'] = recurring_contract_groups.partner_id
```

Comment:
Just in case we would have several results.
```suggestion
            res['partner_id'] = recurring_contract_groups[0].partner_id
```

Reply:
Indeed